picasso/src/main/java/com/squareup/picasso3/BitmapImageDecoder.java

```diff
@@ -0,0 +1,33 @@
+package com.squareup.picasso3;
+
+import android.graphics.BitmapFactory;
+import androidx.annotation.NonNull;
+import java.io.IOException;
+import okio.BufferedSource;
+
+import static com.squareup.picasso3.BitmapUtils.decodeStream;
+
+public final class BitmapImageDecoder implements ImageDecoder {
+
+  @Override public boolean canHandleSource(@NonNull BufferedSource source) {
+    try {
+      if (Utils.isWebPFile(source)) {
+        return true;
+      }
+
+      BitmapFactory.Options options = new BitmapFactory.Options();
+      options.inJustDecodeBounds = true;
+      BitmapFactory.decodeStream(source.peek().inputStream(), null, options);
+      // we successfully decoded the bounds
+      return options.outWidth > 0 && options.outHeight > 0;
+    } catch (IOException e) {
+      return false;
```

Comment:
mmm, i think this whole method should just always true. an I/O problem and all the other things are different from not being able to handle the request.

Reply:
The IOException is just because that's required to be handled, the main idea here is that the stream might not be something Bitmap factory can handle (an SVG, an animated GIF, a fat jpeg, a live photo), so this needs to test that it can decode a bitmap from the stream. 

The fastest way I could think of, without needing to load the entire bitmap into memory, was but simply decoding the bounds.